src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1461,21 +1463,54 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
         $container->setAlias('messenger.transport.encoder', $config['encoder']);
         $container->setAlias('messenger.transport.decoder', $config['decoder']);
 
-        $messageToSenderIdsMapping = array();
-        foreach ($config['routing'] as $message => $messageConfiguration) {
-            $messageToSenderIdsMapping[$message] = $messageConfiguration['senders'];
-        }
+        if (null === $config['default_bus']) {
+            if (count($config['buses']) > 1) {
+                throw new LogicException(sprintf('You need to define a default bus with the "default_bus" configuration. Possible values: %s', implode(', ', array_keys($config['buses']))));
+            }
 
-        $container->getDefinition('messenger.asynchronous.routing.sender_locator')->replaceArgument(1, $messageToSenderIdsMapping);
+            $config['default_bus'] = array_keys($config['buses'])[0];
+        }
 
+        $defaultMiddlewares = array('before' => array('logging'), 'after' => array('route_messages', 'call_message_handler'));
         if ($config['middlewares']['validation']['enabled']) {
```

Comment:
at this point..shouldnt we let users this configure as a regular middleware, per bus, if needed? Extra configs feels a bit too much config IMHO. Or try to prepend either way like 'logging'.

Reply:
Or put different, it's a confusing config as it relies on `default_middlewares` also.

Reply:
This *one* special middleware handling is indeed a little bit weird. Is there any special reason when you would *not* want this middleware? Could it ever cause issues? My instinct is to *always* include it when `default_middlewares` is enabled. If you *really* don't want it for some reason, you can set that config to false (unless there's a somewhat common use-case for not wanting this middleware).

Reply:
I tend to opt-in to a validation middleware.. im not sure validating messages by default makes sense.

Also it's not like adding `- validation` is "bad config" or so =/

Reply:
> Is there any special reason when you would not want this middleware?

Yes, when the component is not included.

> I tend to opt-in to a validation middleware.. im not sure validating messages by default makes sense.

Let's do this. So it is explicit and removes some extra configuration in there.